#820 Item DefaultPrice is synced incorrectly

sworup opened the pull request:
Fixes #820

Set defaultPrice through ItemStoreJoin instead of using item. buy_price which is not correct.

Chris-Petty commented on src/database/DataTypes/ItemStoreJoin.js:
```diff
@@ -9,6 +9,7 @@ export class ItemStoreJoin extends Realm.Object {
     // Make the item no longer visible in this store
     const item = itemResults[0];
     item.isVisible = false;
+    item.defaultPrice = 0;
```
Not sure why this is necessary? Does mSupply desktop do this? Given visibility is controlled server side, I'd have thought the server would send any update to default price anyway?

sworup replied:
This is not necessary, defaultPrice would be set to 0 by Item schema declaration. This code is also doing default value set.  
I did this (I think) for two reasons:
1. I was replicating the isVisible code in line 11.
2. It does not hurt the code and acts as a fail safe

Chris-Petty replied:
Yea I guess if the join is recreated then it'll set the value again. mSupply doesn't actually usually delete these joins anyway, on desktop isVisible is on the join and just switches it True or False.

sworup commented:
@Chris-Petty About `defaultPrice/packSize`: I removed the `parseNumber(record.buy_price) / packSize` code and replaced it with defaultPrice. I do not see any instance of `defaultPrice/packSize` in code.

Chris-Petty commented:
![image](https://user-images.githubusercontent.com/7684221/54564862-c0cca200-4a31-11e9-9cf5-dd1d681c30db.png)

Blocking issue when building dev on to genymotion device that already had a mobile version installed. Cheeky fix would be to make defaultPrice optional again, but I think it's probably better with the default as you have done. So that means bumping the schema version in this PR. Possibly also worth a bit of migration code that sets `defaultPrice` to 0 if currently undefined/null/NaN/falsy

Chris-Petty commented on src/database/DataTypes/ItemStoreJoin.js:
```diff
@@ -9,6 +9,7 @@ export class ItemStoreJoin extends Realm.Object {
     // Make the item no longer visible in this store
     const item = itemResults[0];
     item.isVisible = false;
+    item.defaultPrice = 0;
```
```suggestion
```
I still think it's not correct. 
1. Line 11 is necessary, being the core purpose of this destructor due to how mobile differs from desktop in handling visibility (it's recorded on item, not ItemStoreJoin).
2. If schema is handling it then it's going into redundant zone.

I'm going to push this along so I can merge it.

Chris-Petty commented:
Gah I just butchered this by merging develop into it. Changing base. Derp.
